File: manara/intersection.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function (nums1, nums2) {
  const setA = new Set(nums1);
  const setB = new Set(nums2);

  const result = [];
  for (const num of setA) {
    if (num in setB) result.push(num);
  }

  return result;
};

File: JavaWorkSpace/workspace/src/JavaPlayGround/Intersection.java
package JavaPlayGround;

import java.util.*;

public class Intersection {

    public int[] intersection(int[] nums1, int[] nums2) {
        HashSet<Integer> set1 = new HashSet<Integer>();
        HashSet<Integer> resultSet = new HashSet<Integer>();

        for (int num : nums1) {
            set1.add(num);
        }

        for (int num : nums2) {
            if (set1.contains(num)) {
                resultSet.add(num);
            }
        }

        int[] resultArray = new int[resultSet.size()];
        int index = 0;
        for (int num : resultSet) {
            resultArray[index++] = num;
        }

        return resultArray;

    }

}
